### PR TITLE
Fix ESLint and prettier integration for VS Code

### LINKED ISSUE #20996: ESLint setup in VS Code
There are 2 main ways to lint JS files in our monorepo:

1. You can rely on `pnpm lint-file filename` command to lint a file from your terminal.
2. You can get ESLint suggestions right in your IDE, as you're working on the code.

For some reason, the latter doesn't seem to work in VS Code anymore.

I wonder if it could be related to our switch to PNPM and its workspaces, or to the changes we made in #20085.

#### Steps to reproduce

1. Install VS Code
2. Install [the ESLint extension](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) ( https://github.com/microsoft/vscode-eslint )
3. In a new VS Code window, File > Open... to open the folder where your Jetpack monorepo checkout lives.
4. Go to File > Add Folder to workspace to create a new workspace for Jetpack.
5. Once you've done so, hit Cmd + Shift + P to open the quick dialog, and search for "Preferences Workspace JSON", and open that.
6. There you'll be able to add new ESLint settings for your workspace, inside the settings property. Here is what I have set on my end for example:

```json
"eslint.workingDirectories": [
  "/Users/jeherve/code/jetpack",
  "/Users/jeherve/code/jetpack/tools/js-tools/",
  "/Users/jeherve/code/jetpack/projects/plugins/boost/",
  "/Users/jeherve/code/jetpack/projects/plugins/jetpack/",
  "/Users/jeherve/code/jetpack/projects/plugins/jetpack/extensions/",
  "/Users/jeherve/code/jetpack/projects/plugins/jetpack/modules/search/",
  "/Users/jeherve/code/jetpack/projects/plugins/backup/",
],
"eslint.alwaysShowStatus": true,
"eslint.packageManager": "pnpm",
"eslint.runtime": "node",
"eslint.lintTask.enable": true,
"eslint.format.enable": true,
"eslint.probe": [
    "javascript",
    "javascriptreact",
    "typescript",
    "typescriptreact",
    "html",
    "markdown"
],
```

7. Once you've set that up, hit Cmd + Shift + P again and search for "eslint server"; you'll have the option to restart the server. Do that.
8. At the bottom of the editor, you'll see an output tab; in there, select the ESLint dropdown to get the ESLint output:

image

Here is the output on my end today:

```
[Info  - 5:29:05 PM] ESLint server is starting
[Info  - 5:29:05 PM] ESLint server running in node v16.7.0
[Info  - 5:29:05 PM] ESLint server is running.
[Info  - 5:29:05 PM] ESLint library loaded from: /Users/jeherve/code/jetpack/node_modules/.pnpm/eslint@7.31.0/node_modules/eslint/lib/api.js
Uncaught exception received.
Error: Failed to load config "wpcalypso/react" to extend from.
Referenced from: /Users/jeherve/code/jetpack/.eslintrc.react.js
    at configInvalidError (/Users/jeherve/code/jetpack/node_modules/.pnpm/@eslint+eslintrc@0.4.3/node_modules/@eslint/eslintrc/lib/config-array-factory.js:290:9)
    at ConfigArrayFactory._loadExtendedShareableConfig (/Users/jeherve/code/jetpack/node_modules/.pnpm/@eslint+eslintrc@0.4.3/node_modules/@eslint/eslintrc/lib/config-array-factory.js:883:23)
    at ConfigArrayFactory._loadExtends (/Users/jeherve/code/jetpack/node_modules/.pnpm/@eslint+eslintrc@0.4.3/node_modules/@eslint/eslintrc/lib/config-array-factory.js:781:25)
    at ConfigArrayFactory._normalizeObjectConfigDataBody (/Users/jeherve/code/jetpack/node_modules/.pnpm/@eslint+eslintrc@0.4.3/node_modules/@eslint/eslintrc/lib/config-array-factory.js:720:25)
    at _normalizeObjectConfigDataBody.next (<anonymous>)
    at ConfigArrayFactory._normalizeObjectConfigData (/Users/jeherve/code/jetpack/node_modules/.pnpm/@eslint+eslintrc@0.4.3/node_modules/@eslint/eslintrc/lib/config-array-factory.js:665:20)
    at _normalizeObjectConfigData.next (<anonymous>)
    at ConfigArrayFactory._normalizeObjectConfigDataBody (/Users/jeherve/code/jetpack/node_modules/.pnpm/@eslint+eslintrc@0.4.3/node_modules/@eslint/eslintrc/lib/config-array-factory.js:720:25)
    at _normalizeObjectConfigDataBody.next (<anonymous>)
    at ConfigArrayFactory._normalizeObjectConfigData (/Users/jeherve/code/jetpack/node_modules/.pnpm/@eslint+eslintrc@0.4.3/node_modules/@eslint/eslintrc/lib/config-array-factory.js:665:20)
Uncaught exception received.
```

------

Previous conversations on the topic:

- p1629911000326300-slack-CBG1CP4EN (cc @retrofox )
- p1625223097173000-slack-CBG1CP4EN (cc @brbrr )
- p1627994576016900-slack-C019N9JKKM5 (cc @davidlonjon )

### PR DESCRIPTION
**Pull request by manzoorwanijk**
Fixes #20996

#### Changes proposed in this Pull Request:

* This PR makes `.prettierrc.js` in root directory to be a generic one
* Moves Svelte specific prettier config file to `plugins/boost`
* Fixes ESLint integration by using [@rushstack/eslint-patch](https://www.npmjs.com/package/@rushstack/eslint-patch) as a workaround.

#### Jetpack product discussion

p9dueE-4E8-p2

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

Before following the steps in VS Code, please ensure that you
- Have [VS Code ESLint extension ](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) and [Prettier for VS Code](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) active.
- Have these settings in settings.json (`Cmd + Shift + P` > `Preferences: Open Settings (JSON)`)
```json
  "[javascript][javascriptreact][typescript][typescriptreact]": {
    "editor.formatOnSave": false,
    "editor.defaultFormatter": "esbenp.prettier-vscode"
  },
  "editor.codeActionsOnSave": {
    "source.fixAll.eslint": true,
    "source.fixAll.tslint": true,
    "source.fixAll.stylelint": true
  },
```

##### Steps

* Boot up the PR and run `pnpm install`
* Reload VS Code (`Cmd + Shift + P` > `Developer: Reload Window`)
* Open any file inside `plugins/jetpack/_inc/client/` e.g. `projects/plugins/jetpack/_inc/client/at-a-glance/activity.jsx` 
* Mess up the formatting. For example select some text and run `Outdent line` command (`Cmd + [`) a few times.
* Or add too many empty lines to the file, remove some trailing bracket spaces etc.
* Open problems view (`Cmd + Shift + P` > `Problems: Focus on Problems View`)
* Confirm that you see the problems reported by ESLint.
* Save Changes (`Cmd + S`)
* Confirm that the file is formatted as expected

Screenshot 2022-03-23 at 4 14 22 PM

### CODE REVIEW

File: tools/js-tools/eslintrc/tests.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/**
```
Comment (anomiex) [inline tools/js-tools/eslintrc/tests.js:1]: Why did you add this file but then only use it in one place, leaving the other 16 references to `plugin:jest/recommended` alone?

File: tools/js-tools/package.json [hunk 2]
```diff
@@ -26,11 +29,12 @@
 		"eslint-plugin-es": "4.1.0",
 		"eslint-plugin-es5": "1.5.0",
 		"eslint-plugin-inclusive-language": "2.2.0",
+		"eslint-plugin-import": "2.25.4",
 		"eslint-plugin-jest": "26.0.0",
 		"eslint-plugin-jsdoc": "37.7.0",
 		"eslint-plugin-jsx-a11y": "6.5.1",
 		"eslint-plugin-lodash": "7.3.0",
-		"eslint-plugin-prettier": "4.0.0",
+		"eslint-plugin-prettier": "3.4.1",
```
Comment (anomiex) [inline tools/js-tools/package.json:37]: Why is this being downgraded?

File: tools/js-tools/eslintrc/tests.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+		jest: true,
+	},
+	parserOptions: {
+		ecmaVersion: 2019,
+		ecmaFeatures: {
+			jsx: true,
+		},
+		requireConfigFile: false,
+	},
+	settings: {},
```
Comment (anomiex) [inline tools/js-tools/eslintrc/tests.js:28]: How much of this is actually necessary? Much of it seems redundant to what's in base.js, which should probably be extended alongside of this.

Going by what's being changed in the one file using this file, all that would be needed here is
```suggestion
	extends: [ 'plugin:jest/recommended' ],
```
although based on the other 16 files using `plugin:jest/recommended` you could probably move a few more bits out of that file and into this one.
```suggestion
	extends: [ 'plugin:jest/recommended' ],
	env: { jest: true },
	rules: {
		'jsdoc/check-tag-names': [
			1, // Recommended
			{ definedTags: [ 'jest-environment' ] },
		],
	},
```

**Comment by anomiex** [comment]
> Moves Svelte specific prettier config file to `plugins/boost`

I suppose the question is whether we think svelte will be used anywhere else, or if it'll remain a Boost-only thing.

File: tools/js-tools/package.json (hunk 2 above)
Comment (manzoorwanijk) [reply tools/js-tools/package.json:37]: There was a conflict between different prettier versions and thus ESLint threw this error

`ESLint couldn't determine the plugin "prettier" uniquely.`

Comment (manzoorwanijk) [reply tools/js-tools/package.json:37]: Here is the exact error message
Screenshot 2022-03-24 at 9 42 41 AM

File: tools/js-tools/eslintrc/tests.js (hunk 3 above)
Comment (manzoorwanijk) [reply tools/js-tools/eslintrc/tests.js:28]: Good suggestion. Thank you.

File: projects/plugins/jetpack/extensions/.eslintrc.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 const loadIgnorePatterns = require( '../../../../tools/js-tools/load-eslint-ignore.js' );
 
 module.exports = {
-	extends: [ '../.eslintrc.js', 'plugin:jest/recommended' ],
+	extends: [ '../.eslintrc.js', '../../../../tools/js-tools/eslintrc/tests' ],
```
[the remark this answers is not in the document]
Comment (manzoorwanijk) [reply projects/plugins/jetpack/extensions/.eslintrc.js:4]: Yes, it was already there, thus I didn't remove it.

**Comment by manzoorwanijk** [comment]
> I suppose the question is whether we think svelte will be used anywhere else, or if it'll remain a Boost-only thing.

If we use Svelte anywhere else in future, then we can extract Svelte specific tools to `js-tools` and re-use from there.

File: tools/js-tools/eslintrc/tests.js (hunk 1 above)
Comment (manzoorwanijk) [reply tools/js-tools/eslintrc/tests.js]: Thank you for the catch. I have removed those 16 files and used `overrides` in base eslint config to set `jest/recommended` for the test files.

File: tools/js-tools/package.json [hunk 5]
```diff
@@ -26,11 +29,12 @@
 		"eslint-plugin-es": "4.1.0",
 		"eslint-plugin-es5": "1.5.0",
 		"eslint-plugin-inclusive-language": "2.2.0",
+		"eslint-plugin-import": "2.25.4",
```
[the remark this answers is not in the document]
Comment (manzoorwanijk) [reply tools/js-tools/package.json:32]: The reason is same as that for downgrading `eslint-prettier-plugin`, i.e. there were conflicts when this package was installed inside `plugins/boost`. Thus, I moved it to `js-tools`.

[the remark this answers is not in the document]
Comment (anomiex) [reply tools/js-tools/package.json:32]: My point was that I don't see anywhere in here where it's being used in the first place. And if I delete this line and run `pnpm lint`, nothing breaks.

Also you're going to wind up having to put it back into plugins/boost anyway after #23456, as its a peer dep of `@sveltejs/eslint-config`.

File: tools/js-tools/package.json (hunk 2 above)
Comment (anomiex) [reply tools/js-tools/package.json:37]: Ugh. That's going to be annoying if we have to sync various random plugin versions with whatever `@wordpress/eslint-plugin` happens to include.

I'm tempted to recommend using `.pnpmfile.cjs` to turn `@wordpress/eslint-plugin`'s eslint plugin deps into peer deps instead. That might look like
```diff
diff --git a/.pnpmfile.cjs b/.pnpmfile.cjs
--- a/.pnpmfile.cjs
+++ b/.pnpmfile.cjs
@@ -52,6 +52,16 @@ function fixDeps( pkg ) {
                pkg.dependencies[ 'jest-environment-jsdom' ] = '^27';
        }
 
+       // Turn @wordpress/eslint-plugin's eslint plugin deps into peer deps.
+       if ( pkg.name === '@wordpress/eslint-plugin' ) {
+               for ( const [ dep, ver ] of Object.entries( pkg.dependencies ) ) {
+                       if ( dep.startsWith( 'eslint-plugin-' ) || dep.endsWith( '/eslint-plugin' ) ) {
+                               delete pkg.dependencies[ dep ];
+                               pkg.peerDependencies[ dep ] = ver.replace( /^\^?/, '>=' );
+                       }
+               }
+       }
+
        // Unpin browserslist here.
        if (
                pkg.name === 'react-dev-utils' &&
```

File: tools/js-tools/package.json [hunk 6]
```diff
@@ -17,6 +17,9 @@
 		"@babel/eslint-parser": "7.17.0",
 		"@octokit/auth-token": "2.4.5",
 		"@octokit/rest": "18.6.7",
+		"@rushstack/eslint-patch": "^1.1.1",
+		"@typescript-eslint/eslint-plugin": "^5.16.0",
+		"@typescript-eslint/parser": "^5.16.0",
```
Comment (anomiex) [inline tools/js-tools/package.json:22]: I suspect renovate will come along and want to do
```suggestion
		"@rushstack/eslint-patch": "1.1.1",
		"@typescript-eslint/eslint-plugin": "5.16.0",
		"@typescript-eslint/parser": "5.16.0",
```
here.

File: tools/js-tools/package.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (manzoorwanijk) [reply tools/js-tools/package.json:32]: If I move it back to plugins/boost and run `pnpm install`, here is the error it throws

Screenshot 2022-03-25 at 8 26 53 AM

File: projects/plugins/jetpack/extensions/.eslintrc.js [hunk 7]
```diff
@@ -1,7 +1,7 @@
 const loadIgnorePatterns = require( '../../../../tools/js-tools/load-eslint-ignore.js' );
 
 module.exports = {
-	extends: [ '../.eslintrc.js', 'plugin:jest/recommended' ],
+	extends: [ '../../../../tools/js-tools/eslintrc/jest' ],
 	ignorePatterns: loadIgnorePatterns( __dirname ),
 	env: { jest: true },
```
[the remark this answers is not in the document]
Comment (manzoorwanijk) [reply projects/plugins/jetpack/extensions/.eslintrc.js]: Removed. Thank you.

File: tools/js-tools/package.json (hunk 2 above)
Comment (manzoorwanijk) [reply tools/js-tools/package.json]: I have added the suggestion to `.pnpmfile.cjs`

File: tools/js-tools/package.json (hunk 6 above)
Comment (manzoorwanijk) [reply tools/js-tools/package.json]: Done. Thank you.

File: tools/js-tools/package.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (manzoorwanijk) [reply tools/js-tools/package.json]: After that change you suggested for `.pnpmfile.cjs` to fix dependencies for `@wordpress/eslint-plugin`, it worked. Thank you.

**Comment by anomiex** [comment]
I think we're almost there on this one. A few changes still need to be made:

* Now that we added the `.pnpmfile.cjs` code to make `@wordpress/eslint-plugin`'s eslint plugin deps into peer deps, we get
  ```
  tools/js-tools
  └─┬ @wordpress/eslint-plugin
    └── ✕ missing peer eslint-plugin-import@>=2.25.2
  ```
  So now there is the need for `eslint-plugin-import` in there.
* There are also some missing peer deps in boost:
  ```
  projects/plugins/boost
  └─┬ @sveltejs/eslint-config
    ├── ✕ missing peer @typescript-eslint/eslint-plugin@">= 3"
    └── ✕ missing peer @typescript-eslint/parser@">= 3"
  ```
* Once you fix those peer deps, you'll run into this error:
  ```
  ESLint couldn't determine the plugin "@typescript-eslint" uniquely.

  - /usr/local/src/automattic/jetpack/node_modules/.pnpm/@typescript-eslint+eslint-plugin@5.16.0_aa279973c97bf35b19ed0770e5588c77/node_modules/@typescript-eslint/eslint-plugin/dist/index.js (loaded in "projects/plugins/boost/.eslintrc.cjs")
  - /usr/local/src/automattic/jetpack/node_modules/.pnpm/@typescript-eslint+eslint-plugin@5.16.0_64b579c655ba1f8c39afc110229f83ae/node_modules/@typescript-eslint/eslint-plugin/dist/index.js (loaded in "projects/plugins/boost/.eslintrc.cjs » plugin:@wordpress/recommended")
  ```
  This is because the peer deps don't match between tools/js-tools and plugins/boost. Two possible ways to fix this:
  1. Add a dep on `"eslint": "8.8.0"` to plugins/boost so all the peer deps match.
  2. Move the dep on `@sveltejs/eslint-config` (and `eslint-plugin-svelte3`) to tools/js-tools instead of having Boost load them. That should also let you remove `eslint-plugin-import`, , `@typescript-eslint/eslint-plugin`, and `@typescript-eslint/parser` from plugins/boost. Although I don't know if that'll work for vscode, you'll have to test that part if you decide to try this option.

Yes, all that is redoing some of the stuff you just undid. Sorry, it happens sometimes when trying to get all the pieces to line up right. :(

**Comment by manzoorwanijk** [comment]
> Two possible ways to fix this:
> 1. Add a dep on `"eslint": "8.8.0"` to plugins/boost so all the peer deps match.
> 2. Move the dep on `@sveltejs/eslint-config` (and `eslint-plugin-svelte3`) to tools/js-tools instead of having Boost load them. That should also let you remove `eslint-plugin-import`, , `@typescript-eslint/eslint-plugin`, and `@typescript-eslint/parser` from plugins/boost. Although I don't know if that'll work for vscode, you'll have to test that part if you decide to try this option.

I went with the second option and made the changes. Thank you.

**Review by anomiex — APPROVED** [review]
I note this drops a lot of warnings along the lines of
```
  181:0   warning  Invalid JSDoc @param "options" type "Object"; prefer: "object"             jsdoc/check-types
```
due to setting `.settings.jsdoc.mode` to "typescript". That change seems odd, but I find myself not really caring. If someone else does care, they're welcome to raise the question.

**Comment by anomiex** [comment]
P.S. I went ahead and filed WordPress/gutenberg#39810 asking them to make `@wordpress/eslint-plugin`'s plugin dependencies into peer dependencies upstream.

**Comment by manzoorwanijk** [comment]
> I note this drops a lot of warnings along the lines of
> 
> ```
>   181:0   warning  Invalid JSDoc @param "options" type "Object"; prefer: "object"             jsdoc/check-types
> ```
> 
> due to setting `.settings.jsdoc.mode` to "typescript". That change seems odd

`.settings.jsdoc.mode` has been removed in #23522 because that uses `@typescript-eslint/parser` which uses that option implicitly.

**Comment by jeherve** [comment]
I seem to be able to build the monorepo now; I get the following errors:

```
devDependencies:
+ husky 7.0.4
+ jetpack-cli 1.0.0 <- tools/cli
+ jetpack-js-tools <- tools/js-tools

 ERR_PNPM_PEER_DEP_ISSUES  Unmet peer dependencies

projects/plugins/boost/tests/e2e
└─┬ eslint-plugin-playwright
  └── ✕ missing peer eslint@>=7
Peer dependencies that should be installed:
  eslint@>=7

projects/plugins/jetpack/tests/e2e
└─┬ eslint-plugin-playwright
  └── ✕ missing peer eslint@>=7
Peer dependencies that should be installed:
  eslint@>=7

projects/plugins/search/tests/e2e
└─┬ eslint-plugin-playwright
  └── ✕ missing peer eslint@>=7
Peer dependencies that should be installed:
  eslint@>=7

tools/e2e-commons
└─┬ eslint-plugin-playwright
  └── ✕ missing peer eslint@>=7
Peer dependencies that should be installed:
  eslint@>=7

tools/js-tools
├─┬ eslint-plugin-prettier
│ └── ✕ unmet peer prettier@>=2.0.0: found 2.2.1-beta-1
└─┬ prettier-plugin-svelte
  └── ✕ unmet peer prettier@"^1.16.4 || ^2.0.0": found 2.2.1-beta-1
```

Am I missing something?

**Comment by manzoorwanijk** [comment]
> I don't seem to be able to build the monorepo now; I get the following errors:

May be because of #23456?

**Comment by kraftbj** [comment]
Thanks y'all for working together on this one.

@jeherve I can't duplicate the issue on my end (building `plugins/jetpack` from `master` completes without error).

**Comment by jeherve** [comment]
r242716-wpcom

**Comment by anomiex** [comment]
> I don't seem to be able to build the monorepo now; I get the following errors:

Those shouldn't be a problem thanks to `.pnpm.peerDependencyRules.ignoreMissing` in the monorepo root packahe.json being set to include "eslint". Which version of pnpm are you on? Is there anything else that might be relevant?

**Comment by jeherve** [comment]
I cannot quite tell what version I was running before I updated. I should have checked, sorry!

**Comment by thingalon** [comment]
This change seems to have broken the Jetpack Boost pre-commit hooks:
```
❯ git commit -m "Commit tiny change"
Prettier formatting staged file: projects/plugins/boost/app/assets/src/js/pages/settings/elements/Module.svelte
[error] No parser could be inferred for file: projects/plugins/boost/app/assets/src/js/pages/settings/elements/Module.svelte
node:child_process:903
    throw err;
    ^

Error: Command failed: pnpx prettier --config .prettierrc.js --ignore-path .eslintignore --write projects/plugins/boost/app/assets/src/js/pages/settings/elements/Module.svelte
[error] No parser could be inferred for file: projects/plugins/boost/app/assets/src/js/pages/settings/elements/Module.svelte

    at checkExecSyncError (node:child_process:826:11)
    at execSync (node:child_process:900:15)
    at runPrettier (/Users/thingalon/Projects/jetpack/tools/js-tools/git-hooks/pre-commit-hook.js:165:3)
    at Object.<anonymous> (/Users/thingalon/Projects/jetpack/tools/js-tools/git-hooks/pre-commit-hook.js:433:2)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:17:47 {
  status: 2,
  signal: null,
  output: [
    null,
    Buffer(0) [Uint8Array] [],
    Buffer(125) [Uint8Array] [
       91, 101, 114, 114, 111, 114,  93,  32,  78, 111,  32, 112,
       97, 114, 115, 101, 114,  32,  99, 111, 117, 108, 100,  32,
       98, 101,  32, 105, 110, 102, 101, 114, 114, 101, 100,  32,
      102, 111, 114,  32, 102, 105, 108, 101,  58,  32, 112, 114,
      111, 106, 101,  99, 116, 115,  47, 112, 108, 117, 103, 105,
      110, 115,  47,  98, 111, 111, 115, 116,  47,  97, 112, 112,
       47,  97, 115, 115, 101, 116, 115,  47, 115, 114,  99,  47,
      106, 115,  47, 112,  97, 103, 101, 115,  47, 115, 101, 116,
      116, 105, 110, 103,
      ... 25 more items
    ]
  ],
  pid: 26027,
  stdout: Buffer(0) [Uint8Array] [],
  stderr: Buffer(125) [Uint8Array] [
     91, 101, 114, 114, 111, 114,  93,  32,  78, 111,  32, 112,
     97, 114, 115, 101, 114,  32,  99, 111, 117, 108, 100,  32,
     98, 101,  32, 105, 110, 102, 101, 114, 114, 101, 100,  32,
    102, 111, 114,  32, 102, 105, 108, 101,  58,  32, 112, 114,
    111, 106, 101,  99, 116, 115,  47, 112, 108, 117, 103, 105,
    110, 115,  47,  98, 111, 111, 115, 116,  47,  97, 112, 112,
     47,  97, 115, 115, 101, 116, 115,  47, 115, 114,  99,  47,
    106, 115,  47, 112,  97, 103, 101, 115,  47, 115, 101, 116,
    116, 105, 110, 103,
    ... 25 more items
  ]
}
husky - pre-commit hook exited with code 1 (error)
```

I'm on pnpm 6.32.3:
```
projects/plugins/boost on  text/pre-commit [+] is 📦 v1.4.2-alpha via  v16.13.2 via 🐘 v8.0.15 
❯ pnpm -v
6.32.3
```

I'll try to figure out how / why, but any help appreciated :)

**Comment by thingalon** [comment]
I've found the issue -- the pre-commit hooks explicitly use the root `.prettierrc.js` file, breaking checks for svelte files in Jetpack Boost.

I think I have a fix here to be reviewed: https://github.com/Automattic/jetpack/pull/23691

**Pull request merged**